### DISCUSSION TITLE
ArgoCD High Available Setups

### DISCUSSION THREAD

**Discussion by Josephred999** [argoproj/argo-cd#15177] [Ideas]
Hello Argo community,

I am trying to setup a High-Availability for the ArgoCD Setups, but I see the only option is Argo's own HA Setup will provide multiple Redis instances and spread across multiple Kubernetes Nodes.

Other than that, are there any possibilities for setting up Active-Active or Active-Passive setups where we can move the Redis component out of the ArgoCD setups and put it behind a load-balancer and consume it in the ArgoCD instances??

Has anyone achieved this level of High Availability achieved in our community?

Looking for ideas, thoughts, quick-comments, one-liners and even meme's

**Comment by FernandoMiguel** [comment]
you mean , something like redis external?
https://github.com/argoproj/argo-helm/blob/7dcc174402d4a81dcdd6324bd7be095a3c52a921/charts/argo-cd/values.yaml#L1428

**Reply by FernandoMiguel** [reply]
@Josephred999

**Comment by vkg23** [comment]
'+ 1

There is an extensive use of Argo for the CD across multiple dev's and clusters in my case. 

Though the Argo could be installed in HA mode ,  what are the possibilities of reducing impact , due to version upgrade of running ARGO setup etc / other downtime's. 

was thinking , having an active/active/passive setup helps to have the seamless switching .  
OR is there a way to switch a new setup of argo with the same old backend etc

**Comment by rumstead** [comment]
Maybe this past discussion can spark some ideas. 

https://github.com/argoproj/argo-cd/discussions/9488
